python/pyspark/pandas/tests/data_type_ops/test_boolean_ops.py

```diff
@@ -228,9 +230,272 @@ def test_rmod(self):
         self.assertRaises(TypeError, lambda: datetime.date(1994, 1, 1) % self.psser)
         self.assertRaises(TypeError, lambda: True % self.psser)
 
+    def test_and(self):
+        pser = pd.Series([True, False, None], dtype='bool')
+        psser = ps.from_pandas(pser)
+        self.assert_eq(pser & True, psser & True)
+        self.assert_eq(pser & False, psser & False)
+        self.assert_eq(pser & pser, psser & psser)
+
+        other_pser = pd.Series([False, None, True], dtype='bool')
+        other_psser = ps.from_pandas(other_pser)
+        with option_context("compute.ops_on_diff_frames", True):
+            self.assert_eq(pser & other_pser, psser & other_psser)
+            self.check_extension(
+                pser & other_pser.astype('boolean'), psser & other_psser.astype('boolean'))
+            self.assert_eq(other_pser & pser, other_psser & psser)
+
+    def test_rand(self):
+        pser = pd.Series([True, False, None], dtype='bool')
+        psser = ps.from_pandas(pser)
+        self.assert_eq(True & pser, True & psser)
+        self.assert_eq(False & pser, False & psser)
+
+    def test_or(self):
+        pser = pd.Series([True, False, None], dtype='bool')
+        psser = ps.from_pandas(pser)
+        self.assert_eq(pser | True, psser | True)
+        self.assert_eq(pser | False, psser | False)
+        self.assert_eq(pser | pser, psser | psser)
+        self.assert_eq(True | pser, True | psser)
+        self.assert_eq(False | pser, False | psser)
+
+        other_pser = pd.Series([False, None, True], dtype='bool')
+        other_psser = ps.from_pandas(other_pser)
+        with option_context("compute.ops_on_diff_frames", True):
+            self.assert_eq(pser | other_pser, psser | other_psser)
+            self.check_extension(
+                pser | other_pser.astype('boolean'), psser | other_psser.astype('boolean'))
+            self.assert_eq(other_pser | pser, other_psser | psser)
+
+    def test_ror(self):
+        pser = pd.Series([True, False, None], dtype='bool')
+        psser = ps.from_pandas(pser)
+        self.assert_eq(True | pser, True | psser)
+        self.assert_eq(False | pser, False | psser)
+
+
+@unittest.skipIf(
+    not extension_dtypes_available, "pandas extension dtypes are not available"
+)
+class BooleanExtensionOpsTest(BooleanOpsTest, PandasOnSparkTestCase, TestCasesUtils):
```

Comment:
Seems like we don't reuse tests from `BooleanOpsTest` at all. We should remove it for now?

Reply:
Makes sense! Removed.